packages/office-ui-fabric-react/src/components/Image/Image.tsx

```diff
@@ -137,11 +138,9 @@ export class Image extends React.Component<IImageProps, IImageState> {
   private _computeCoverStyle(props: IImageProps) {
     let { imageFit } = props;
     if (imageFit === ImageFit.cover || imageFit === ImageFit.contain) {
-      let { image } = this.refs;
+      let { image, frame } = this.refs;
       if (image) {
-        let { width, height } = props;
-
-        let desiredRatio = (width as number) / (height as number);
+        let desiredRatio = frame.clientWidth / frame.clientHeight;
```

Comment:
Good point. I'll add a check to prefer the width and height props, and only measure the element if those aren't present.

At some point, we'll also want to respond to changes in window size (in case the frame's aspect ratio has changed and we need a different cover style) and performance will be an even bigger concern there.